dags/miovision_pull.py

```diff
@@ -0,0 +1,190 @@
+            postgres_conn_id='miovision_api_bot',
+            autocommit=True
+        )
+
+        check_jan_1st.override(task_id="check_annual_partition")() >> create_annual_partition >> (
+            check_1st_of_month.override(task_id="check_month_partition")() >> create_month_partition
+        )
+
+    @task(trigger_rule='none_failed', retries = 1)
+    def pull_miovision(ds = None, **context):
+        if context["params"]["intersection"] == [0]:
+            INTERSECTION = ()
+        else:
+            INTERSECTION = tuple(context["params"]["intersection"])
+        run_api(start_date=ds,
+                end_date=ds_add(ds, 1),
+                path=API_CONFIG_PATH,
+                intersection=INTERSECTION,
+                pull=True,
+                dupes=True)
+
+    @task_group
+    def miovision_agg():
+        @task
+        def find_gaps_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                find_gaps(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_15_min_mvt_task(ds = None, **context):
+            mio_postgres = PostgresHook("miovision_api_bot")
+            if context["params"]["intersection"] != 0:
+                intersections = get_intersection_info(conn, intersection=context["params"]["intersection"])
+                with mio_postgres.get_conn() as conn:
+                    aggregate_15_min_mvt(conn, ds, ds_add(ds, 1), True, intersections)
+            else:
+                with mio_postgres.get_conn() as conn:
+                    aggregate_15_min_mvt(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_15_min_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                aggregate_15_min(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_volumes_daily_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                aggregate_volumes_daily(conn, ds, ds_add(ds, 1))
+
+        @task
+        def get_report_dates_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                get_report_dates(conn, ds, ds_add(ds, 1))
+
+        find_gaps_task() >> aggregate_15_min_mvt_task() >> [aggregate_15_min_task(), aggregate_volumes_daily_task()]
+        get_report_dates_task()
+
+    t_done = ExternalTaskMarker(
+            task_id="done",
+            external_dag_id="miovision_check",
+            external_task_id="starting_point"
+    )
+
+    @task_group()
+    def data_checks():
+        """Task group to check critical data quality measures which could warrant re-running the DAG."""
```

Comment:
apparently we can add this as a tooltip within task_group() so the description shows up in the UI

Reply:
Trying!